src/main/groovy/org/jenkinsci/plugins/pipeline/modeldefinition/Utils.groovy

```diff
@@ -57,6 +56,14 @@ import java.lang.reflect.ParameterizedType;
 @SuppressFBWarnings(value="SE_NO_SERIALVERSIONID")
 public class Utils {
 
+    public static Map<String,String> blockedStepsBase() {
+        return [
+            "stage":      'The stage step cannot be used in Declarative Pipelines',
```

Comment:
Does it mean that we cannot have conditional stages in Declarative pipeline? E.g. "publish when a user sets up the Publish checkbox". Smells like a design 🐛 for me